packages/client/hmi-client/src/types/common.ts

```diff
@@ -212,3 +213,25 @@ export interface NotificationItem {
 	lastUpdated: number;
 	acknowledged: boolean;
 }
+
+export const ProgrammingLanguageVersion: { [key in ProgrammingLanguage]: string } = {
+	[ProgrammingLanguage.Python]: 'python3',
+	[ProgrammingLanguage.R]: 'ir',
+	[ProgrammingLanguage.Julia]: 'julia-1.10',
+	[ProgrammingLanguage.Zip]: 'zip'
+};
+
+/**
+ * Returns an array of options for programming languages.
+ * Each option is an object with a `name` property that is a `ProgrammingLanguage` and a `value` property that is the corresponding version string.
+ * The `Zip` programming language is excluded from the options.
+ * @returns {Array} An array of options for programming languages.
+ */
+export const programmingLanguageOptions = (): { name: string; value: string }[] =>
+	Object.values(ProgrammingLanguage)
+		.filter((lang) => lang !== ProgrammingLanguage.Zip)
+		.map((lang) => ({
+			name:
+				lang && `${lang[0].toUpperCase() + lang.slice(1)} (${ProgrammingLanguageVersion[lang]})`,
+			value: ProgrammingLanguageVersion[lang]
+		}));
```

Comment:
This feels like an awkward construct, wouldn't it be easier if we start off by storing the programming languages and their versions together? Then we don't need to do these kind of data wrangling and have the slight problem of independently maintaining "languages" across two physical locations in the repo.

Just food for thought for later - no need to actionnow.

Reply:
might be good to use the beaker api to fetch the languages and version to use in the dropdown in the future: `https://beaker.terarium.ai/contexts`